clang/include/clang/AST/DependenceFlags.h

```diff
@@ -49,7 +49,7 @@ struct ExprDependenceScope {
 using ExprDependence = ExprDependenceScope::ExprDependence;
 
 struct TypeDependenceScope {
-  enum TypeDependence : uint8_t {
+  enum TypeDependence : unsigned {
```

Comment:
Can you explain this change? Making Type bigger than it is can have a pretty large impact on the ressource need of clang.

Reply:
The only place this enum is stored is bit-field I annotate in this patch. Like it was with `TypeAlignment`, I'm only encoding status quo that actual underlying type for this enum is unsigned. There is no impact on memory footprint, because I'm yet to see us deriving bit-field width from underlying type of enum.

Reply:
In this particular case, width of bit-field is not hardcoded, but derived from enumerators via `llvm::BitWidth<TypeDependence>`

Reply:
I had a chat with @Endilll offline.
The reason for this change is that `preferred_type` (#69104) warns when the enumerator has a different underlying type as the bit field. https://godbolt.org/z/zz99s6M3j

This warning does seem a bit off to me - but I really like its intent.
The warning only triggers on enumerators (which probably make sense as a general warning may have too many false positives), but I think the warning is too restrictive.
If the underlying type has the same signedness and has a size equal to **or smaller**  than the the size of the type of the bitfield, I don't think a warning should be produced.

I have convinced myself that the changes here are most likely fine (`TypeDependence` is (afaict) only stored in TypeBitfields, as far as i can tell, and this uses the number of bits of the greatest enumerator, which i did not initially realize).

Yet I do not like the idea that we would change the underlying type of all enums used in bitfields for the sake of that warning.

My preferred solution would be to make `-Wbitfield-type` less aggressive and not change the underlying types of enums is this patch.

Reply:
@erichkeane @AaronBallman wdyt?

Reply:
>>My preferred solution would be to make -Wbitfield-type less aggressive and not change the underlying types of enums is this patch.

Yeah, this I think is my preference.

Reply:
Now that we're reviewing https://github.com/llvm/llvm-project/pull/70632 should we revert the changes to underlying types from this patch?

Reply:
Yeah, I'm putting this on hold for a bit, and refactor enumerations in NFC commits, so that this PR lands without "holes" because some enums are not complete at the point where I put `preferred_type`. After that underlying type changes this PR does shouldn't be necessary anymore.

Reply:
I reverted changes to underlying types. This should be good to go now.